tcf_website/api/filters.py

```diff
@@ -1,23 +1,21 @@
 """"Custom DRF pagination classes"""
-from django_filters import FilterSet, BooleanFilter, NumberFilter
+from django_filters import FilterSet, NumberFilter
 from ..models import Instructor, Semester
 
 # pylint: disable=unused-argument,no-self-use
 
 
 class InstructorFilter(FilterSet):
     """Filter for Instructor"""
-    section__course = NumberFilter(field_name='section__course', distinct=True)
-    recent = BooleanFilter(method='filter_recent', distinct=True)
+    course = NumberFilter(method='filter_recent_instructors_by_course')
 
-    def filter_recent(self, queryset, name, value):
-        """Filter method for most recent 5 years"""
-        if value is True:
-            return queryset.filter(
-                section__semester__year__gt=Semester.latest().year - 5,
-            )
-        return queryset
+    def filter_recent_instructors_by_course(self, queryset, name, value):
+        """Filters instructors who taught the course in the recent 5 years"""
+        return queryset.filter(
+            section__course=value,
+            section__semester__year__gte=Semester.latest().year - 5,

```

Comment:
Small nitpick — `gte` here will actually get all courses in the past 6 years (if you're including the current year as one of the years); the old one used `gt`

Reply:
I didn't make this decision myself--I just inherited the business logic from previous code. 
https://github.com/thecourseforum/theCourseForum2/commit/93a599114e4205bec2c4b5f4a373120c96edda50#diff-2c0a59c23c4096bcb34212e25b50c2bd9cae3d64dfb4065c08648f897bd2e3e5R44-R47).
Also, right now the latest semester is Spring 2021, which is the next year. Maybe the execs can discuss this (which was [brought up in Discord as well](https://discord.com/channels/752270432329269328/752272114144509995/773019426840117298)) and let everyone know?

Reply:
Interestingly enough, the old version of this (that's currently in `dev`) is indeed using `gt` ... we should definitely sort this out at some point

![image](https://user-images.githubusercontent.com/55100084/100065715-21fd6900-2e02-11eb-9e14-dc69d933571a.png)

Reply:
Yeah, I wrote the code and the left, and I "fixed" it in https://github.com/thecourseforum/theCourseForum2/pull/215/commits/f39317e0542c496730c002e4da4eb636d058e0b4 to be consistent with the rest of the code base.

One more thing to note, we need to decide whether we want to compare by year or the number of semesters.
- When the latest semester is Fall 2021, the most recent 5 years will be from January 2016 (or 2017) to Fall 2021. 
- When the latest semester is Spring 2021, the most recent 5 years will be from January 2016 (or 2017) to Spring 2021. 

So we see one (or two?) more semester when the latest semester is Fall. However, the code becomes much more complicated if we use the number of semesters instead of years.